mtags/src/main/scala-3/scala/meta/internal/pc/completions/CompletionSuffix.scala

```diff
@@ -0,0 +1,48 @@
+package scala.meta.internal.pc.completions
+
+/**
+ * @param brace should we add "()" suffix?
+ * @param bracket should we add "[]" suffix?
+ * @param template should we add "{}" suffix?
+ * @param snippet which suffix should we insert the snippet $0
+ */
+case class CompletionSuffix(
+    brace: Boolean,
+    bracket: Boolean,
+    template: Boolean,
+    snippet: SuffixKind,
+):
+  def hasSnippet = snippet != SuffixKind.NoSuffix
+  def map(copyFn: CompletionSuffix => CompletionSuffix) = copyFn(this)
+  def toEdit: String =
+    if !hasSuffix then ""
+    else
+      val braceSuffix =
+        if brace && snippet == SuffixKind.Brace then "($0)"
+        else if brace then "()"
+        else ""
+      val bracketSuffix =
+        if bracket && snippet == SuffixKind.Bracket then "[$0]"
+        else if bracket then "[]"
+        else ""
+      val templateSuffix =
+        if template && snippet == SuffixKind.Template then " {$0}"
```

Comment:
Is it expected to get all the suffixes at the same time like for example `Foo[](){}` when we have `trait Foo[T](a: Int)`?

Reply:
Yes!

Reply:
`Foo[$0]($0){$0} ` would not work though, we would need `Foo[$0]($1){$2} ` which would mean that users could jump with tab to next positions, we shouldn't have multiple `$0`

Reply:
As `CompletionSuffix` has an enum for where to put the `$0`, there shouldn't be a case like `Foo[$0]($0){$0}`.

Reply:
ach ok, it might be an improvement to do `$0,$1` etc.